packages/modelviewer.dev/src/docs-and-examples/create-html.ts

```diff
@@ -0,0 +1,238 @@
+/* @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface Entry {
+  name: string, htmlName: string, description: string, default: string[],
+      links: string[],
+}
+
+interface Category {
+  Title: string, Attributes: Entry[], 'CSS Custom Properties': Entry[],
+      Properties: Entry[], Methods: Entry[], Events: Entry[], Slots: Entry[],
+}
+
+const CategoryConstant: Category = {
+  Title: '',
+  Attributes: [],
+  'CSS Custom Properties': [],
+  Properties: [],
+  Methods: [],
+  Events: [],
+  Slots: [],
+}
+
+function createSubcategorySidebar(subcategory: string, lowerCaseTitle: string):
+    string {
+      const lowerCaseKey = getLowerCaseKey(subcategory);
+      return `
+    <div class="subCategory" id=${'subCategory'.concat(subcategory)}>
+      <h4 class="subcategory-header" id=${
+          lowerCaseTitle.concat('-', lowerCaseKey, '-sidebar')}>
+        <a class="darken" href="#${lowerCaseTitle.concat('-', lowerCaseKey)}">${
+          subcategory}</a>
+      <h4>
+    </div>
+  `;
+    }
+
+function createSidebar(category: Category, docsOrExamples: 'docs'|'examples') {
+  if (docsOrExamples === 'examples')
+    return;
+  const container = document.getElementById('sidebar-category-container');
+  const lowerCaseTitle = category.Title.toLowerCase();
+  let subcategories = Object.keys(category);
+  subcategories = subcategories.filter(k => k !== 'Title');
+
+  const categoryContainer = `
+    <div class="category" id=${lowerCaseTitle.concat('aboveHeader')}>
+      <h3 id=${lowerCaseTitle.concat('-sidebar')}>
+        <a class="darken" href="#${lowerCaseTitle}">${category.Title}</a>
+      <h3>
+    </div>
+  `;
+
+  container!.innerHTML += categoryContainer;
+
+  const innerCategory =
+      document.getElementById(lowerCaseTitle.concat('aboveHeader'));
+  for (const subcategory of subcategories) {
+    innerCategory!.innerHTML +=
+        createSubcategorySidebar(subcategory, lowerCaseTitle);
+
+    const innerSubcategory =
+        document.getElementById(lowerCaseTitle.concat('aboveHeader'));
+    const lowerCaseKey = getLowerCaseKey(subcategory);
+    const entries =
+        (<Entry[]>category[subcategory as keyof typeof CategoryConstant]);
+    for (const entry of entries) {
+      innerSubcategory!.innerHTML += `
+        <div class="element de-active" id=${
```

Comment:
Generally with raw string text blocks I restart the indenting at zero, as those tabs are actually part of the string, but don't have much meaning out of context. It also helps them stand out and not line wrap so heavily.

Reply:
done